modules/phase_field/include/kernels/ACPFFrac.h

```diff
@@ -0,0 +1,68 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef ACPFFRAC_H
+#define ACPFFRAC_H
+/**
+ * Phase field based fracture model
+ * This kernel computes the residual and jacobian for bulk free energy contribution to c
+ * Refer to Formulation: Miehe et. al., Int. J. Num. Methods Engg., 2010, 83. 1273-1311 Equation 63
+ */
+#include "Kernel.h"
+#include "RankTwoTensor.h"
+#include "JvarMapInterface.h"
+#include "DerivativeMaterialInterface.h"
+
+//Forward Declarations
+class ACPFFrac;
+
+template<>
+InputParameters validParams<ACPFFrac>();
+
+class ACPFFrac : public DerivativeMaterialInterface<JvarMapKernelInterface<Kernel> >
```

Comment:
I'd strongly prefer a more verbose class name. We have enough of cryptic classes already. `AllenCahnPFFracture` maybe? ...Wait, does `AC` even stand for Allen-Cahn? Or is it `AnisotropicCracking`? ... case and point.